runtime/common/src/transfer_filter.rs

```diff
@@ -13,32 +13,49 @@
 use cfg_primitives::{AccountId, Balance};
 use cfg_traits::{PreConditions, TransferAllowance};
 use cfg_types::{domain_address::DomainAddress, locations::Location, tokens::CurrencyId};
+use frame_support::{dispatch::TypeInfo, traits::IsSubType};
 use pallet_restricted_tokens::TransferDetails;
 use pallet_restricted_xtokens::TransferEffects;
-use parity_scale_codec::Encode;
+use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
 use sp_core::Hasher;
 use sp_runtime::{
-	traits::{BlakeTwo256, Convert},
+	traits::{BlakeTwo256, Convert, DispatchInfoOf, SignedExtension, StaticLookup},
+	transaction_validity::{InvalidTransaction, TransactionValidityError},
 	DispatchError, DispatchResult, TokenError,
 };
 use xcm::v3::{MultiAsset, MultiLocation};
 
+#[derive(
+	Clone, Copy, PartialOrd, Ord, PartialEq, Eq, Debug, Encode, Decode, TypeInfo, MaxEncodedLen,
+)]
+pub enum FilterCurrency {
+	All,
+	Specific(CurrencyId),
+}
+
```

Comment:
I am still confusing about the second point: 

> If the first is false, and then we check for `FilterCurrency::All`. Is it not the same as just checking it for `FilterCurrency::All`?

Maybe would help adding docs here?

Reply:
where did I write that line? ^^

Reply:
Logically, you can set
* an allowance for an account to be allowed for ALL currencies
* an allowance for an account to be allowed to ONE currency

There might be an overlap if you set account `A` both for `All` and for `Specific(CURRENCY)`. Do you want to prevent that? 
Preventing the other way around is harder, as It would need to iterate over the whole storage of that account.

Reply:
> where did I write that line? ^^

I wrote it 😆

Ok, I think I follow you now 👍🏻, thanks!